scripts/collect.py

```diff
@@ -0,0 +1,80 @@
+import argparse
+import os
+import shutil
+
+os.environ["OMP_NUM_THREADS"] = "1"
+import numpy as np
+import cv2
+
+from models.ddpg.model import load_model
+from utils.env import launch_env
+from utils.util import set_seeds, parse_config
+from utils.reward_shaping.env_utils import Transformer, PreliminaryTransformer
+
+
+def ask_to_remove_if_exists(directory):
+    if os.path.exists(directory):
+      should_remove = input(f'The directory {directory} exists. Should it be removed? (y/n): ')
+      if should_remove.lower() == 'y':
+        shutil.rmtree(directory)
+
+
+def collect(model_directory, output_dir, episodes, max_steps):
+    config = parse_config(directory=model_directory)
+    global_seed = config['training']['global_seed']
+    set_seeds(global_seed)
+    model = load_model(model_directory, load_gpu_model_on_cpu=True)
+
+    ask_to_remove_if_exists(output_dir)
+    os.makedirs(output_dir, exist_ok=True)
+
+    env = launch_env()
+    env_seed = config['environment']['core'].get('seed', global_seed)
+    env.seed(env_seed)
+    preprocessor = PreliminaryTransformer()
+    transformer = Transformer()
+
+    total_samples = 0
+    for episode in range(episodes):
+        print('resetting environment')
+        raw_observation = env.reset()
+        preprocessed_observation = preprocessor.transform(raw_observation)
+        transformer.reset(preprocessed_observation)
+        transformed_observation = transformer.transform(preprocessed_observation)
+        reward_sum = 0.0
+        for step in range(max_steps):
+            action = model.act(transformed_observation)
+            print('action', action)
+
+            sample_filename = f'ep_{str(episode).zfill(2)}_{str(step).zfill(3)}'
```

Comment:
очень и очень сильно nitpicking, но т.к. gym поддерживает только python ≥ 3.6, то можно вот так:
```suggestion
            sample_filename = f'ep_{episode:02}_{step:03}'
```
https://stackoverflow.com/a/339013

Reply:
Действительно, гораздо удобнее.